data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java

```diff
@@ -184,7 +187,10 @@ public void setUp() {
                 .thenReturn(authenticationProvider);
 
         testBuffer = getBuffer();
-        HTTPSourceUnderTest = new HTTPSource(sourceConfig, pluginMetrics, pluginFactory);
+        pipelineDescription = new PluginSetting("http_source", Collections.emptyMap()) {{
```

Comment:
Using `{{`  creates an anonymous class and is generally considered an anti-pattern.

This is better:
```
pipelineDescription = new PluginSetting("http_source", Collections.emptyMap());
pipelineDescription.setPipelineName(TEST_PIPELINE_NAME);
```

Also, why not use a mock?
```
pipelineDescription = mock(PipelineDescription.class);
when(pipelineDescription.getPipelineName).thenReturn(TEST_PIPELINE_NAME);
```